controllers/managedosimage_controller.go

```diff
@@ -505,16 +504,16 @@ func (r *ManagedOSImageReconciler) objToFleetBundleResources(objs []runtime.Obje
 }
 
 func getCloudConfig(managedOSImage *elementalv1.ManagedOSImage) ([]byte, error) {
-	if managedOSImage.Spec.CloudConfig == nil || len(managedOSImage.Spec.CloudConfig.Data) == 0 {
+	if len(managedOSImage.Spec.CloudConfig) == 0 {
 		return []byte{}, nil
 	}
 
-	data, err := yaml.Marshal(managedOSImage.Spec.CloudConfig.Data)
+	data, err := util.MarshalCloudConfig(managedOSImage.Spec.CloudConfig)
 	if err != nil {
-		return nil, fmt.Errorf("failed to marshal cloud config: %w", err)
+		return nil, fmt.Errorf("mashalling cloud config: %w", err)
 	}
 
-	return append([]byte("#cloud-config\n"), data...), nil
+	return data, nil
```

Comment:
I am wondering how is this then passed to the host without the `#cloud-config` shebang. Isn't it applied as a configuration file to the host? If so, this is needed to let yip know this should be interpreted with cloud-init syntax.

Reply:
`util.MarshalCloudConfig` does it. It's the same util function we use for the registration.